addons/l10n_eg_edi_eta/models/account_edi_format.py

```diff
@@ -0,0 +1,437 @@
+        except (ValueError, requests.exceptions.ConnectionError, requests.exceptions.MissingSchema, requests.exceptions.Timeout, requests.exceptions.HTTPError) as ex:
+            return {
+                'error': _('Please try again later. Error: \n%s', ex),
+                'blocking_level': 'warning'
+            }
+        if response_data.get('error'):
+            return {
+                'error': response_data.get('error', _('Unknown error')),
+                'blocking_level': 'error'
+            }
+        return response_data.get('access_token') or {
+            'error': _('Unable to retrieve ETA token'),
+            'blocking_level': 'warning'
+        }
+
+    @api.model
+    def _l10n_eg_get_einvoice_status(self, invoice):
+        token = self._l10n_eg_get_einvoice_token(invoice)
+        api_domain = self._l10n_eg_get_eta_api_domain()
+        if isinstance(token, dict) and token.get('error', False):
+            return {invoice: token}
+        request_url = '%s/api/v1.0/documentsubmissions/%s' % (api_domain, url_quote(invoice.l10n_eg_submission_id))
+
+        headers = {'Content-Type': 'application/json',
+                   'Authorization': 'Bearer %s' % token}
+        try:
+            request_response = requests.get(
+                request_url, headers=headers, timeout=(5, 10))
+            submission_data = request_response.json()
+        except (ValueError, requests.exceptions.ConnectionError, requests.exceptions.MissingSchema, requests.exceptions.Timeout, requests.exceptions.HTTPError) as ex:
+            return {invoice: {'error': str(ex), 'blocking_level': 'warning'}}
+        if submission_data.get('error'):
+            return {invoice: {
+                    'error': submission_data.get('error'),
+                    'blocking_level': 'error'
+                }}
+        document_summary = [doc for doc in submission_data.get('documentSummary', []) if doc.get('uuid') == invoice.l10n_eg_uuid]
+        return_dict = {
+            'Invalid':  {
+                'error':  _("This invoice has been marked as invalid by the ETA, Please check the ETA website for more information"),
+                'blocking_level': 'error'
+            },
+            'Submitted': {
+                'error':  _("This invoice has already been sent to the ETA and is awaiting validation"),
+                'blocking_level': 'info'
+            },
+            'Valid': {'success': True},
+            'Cancelled': {'error': _('Document Cannceled'), 'blocking_level': 'error'},
+        }
+        if document_summary and return_dict.get(document_summary[0].get('status')):
+            return {invoice: return_dict.get(document_summary[0]['status'])}
+        return {invoice: {'error': _('Unknow error has occured'), 'blocking_level': 'warning'}}
+
+    def _l10n_eg_validate_different_vats(self, invoice):
+        return invoice.journal_id.l10n_eg_branch_id.vat != invoice.partner_id.vat
+
+    def _l10n_eg_validate_tax_codes(self, invoice):
```

Comment:
It is of no help to have these checks in separate methods.  Better to put the code directly in the method.